### PR TITLE
feat(page): add page param to support pagination without knowing dids

### PR DESCRIPTION
**Pull request by Avantol13**
### New Features
- add `page` param to list records endpoint to support pagination without using a did

### CODE REVIEW

File: indexd/index/drivers/alchemy.py [hunk 1]
```diff
@@ -432,14 +433,26 @@ def ids(
...
+            if page is not None:
+                # when we're paginating, order by updated date so newly added stuff is
+                # at the end (reduce risk that a new records ends up in a page
+                # earlier on) and allows for some logic to check for newly added records
+                # (e.g. parallelly processing from beginning -> middle and ending -> middle
+                #       and as a final step, checking the "ending"+1 to see if there are
+                #       new records).
+                query = query.order_by(IndexRecord.updated_date)
+            else:
+                query = query.order_by(IndexRecord.did)
```
Comment (paulineribeyre) [inline indexd/index/drivers/alchemy.py]: should we just always order by date?

File: indexd/index/drivers/alchemy.py [hunk 2]
```diff
@@ -432,14 +433,26 @@ def ids(
...
 
-            query = query.order_by(IndexRecord.did)
+            if page is not None:
+                # when we're paginating, order by updated date so newly added stuff is
+                # at the end (reduce risk that a new records ends up in a page
+                # earlier on) and allows for some logic to check for newly added records
+                # (e.g. parallelly processing from beginning -> middle and ending -> middle
+                #       and as a final step, checking the "ending"+1 to see if there are
+                #       new records).
+                query = query.order_by(IndexRecord.updated_date)
```
Comment (paulineribeyre) [inline indexd/index/drivers/alchemy.py]: why not order by decreasing date? if we use this to validate after indexing for example, we could just get the first pages

File: indexd/index/drivers/alchemy.py (hunk 1 above)
Comment (Avantol13) [reply indexd/index/drivers/alchemy.py]: I thought about that... probably? I was afraid to change the existing behavior tbh, but I guess it shouldn't matter, you'll always get the next `did` regardless of the ordering

File: indexd/index/drivers/alchemy.py (hunk 2 above)
Comment (Avantol13) [reply indexd/index/drivers/alchemy.py]: The way this is set the dates later in the past will be first such that any additional records end up at the end. This means that existing pages won't change content, whereas if we were to do the opposite, a new record would bump everything else forward, changing the content on all existing pages. So I didn't want to break things mid-processing. If you're curious, this is the logic I was cross-testing with gen3 sdk: https://github.com/uc-cdis/gen3sdk-python/blob/907d418ebcf46ea345472229982c604c54cce179/gen3/tools/indexing.py#L82

Comment (paulineribeyre) [reply indexd/index/drivers/alchemy.py]: makes sense!! thanks

**Pull request merged**